src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java

```diff
@@ -328,6 +331,12 @@ private int instances(AppDeploymentRequest request) {
 		if (route(request) != null) {
 			manifest.route(Route.builder().route(route(request)).build());
 		}
+		if (! routes(request).isEmpty()){
```

Comment:
Can we add assertions here to check if this property being mutually exclusive with host and domain (if they are present)?

Reply:
We could, but that is already handled in the underlying cf client:
https://github.com/cloudfoundry/cf-java-client/blob/dc168fdbcbbb72798d78e555acb72fc6a37ba9c3/cloudfoundry-operations/src/main/java/org/cloudfoundry/operations/applications/_ApplicationManifest.java#L36-L49